sled-agent/src/common/underlay.rs

```diff
@@ -21,28 +29,44 @@ pub enum Error {
 
     #[error(transparent)]
     BadAddrObj(#[from] addrobj::ParseError),
+
+    #[error("Could not determine if host is a Gimlet: {0}")]
+    SystemDetection(#[from] anyhow::Error),
```

Comment:
This `#[from]` feels a little dicey to me because `anyhow::Error` is so generic - now any function that returns an `Error` but calls a function returning `anyhow::Error` can use `?`, and it will automatically become `Error::SystemDetection`. I think I'd change this to `#[source]` and use explicit `.map_err(Error::SystemDetection)` calls.